caddyhttp/markdown/setup.go

```diff
@@ -115,28 +117,48 @@ func loadParams(c *caddy.Controller, mdc *Config) error {
 			fpath := filepath.ToSlash(filepath.Clean(cfg.Root + string(filepath.Separator) + tArgs[0]))
 
 			if err := SetTemplate(mdc.Template, "", fpath); err != nil {
-				c.Errf("default template parse error: %v", err)
+				return c.Errf("default template parse error: %v", err)
 			}
+
+			mdc.TemplateFiles[""] = fpath
 			return nil
 		case 2:
 			fpath := filepath.ToSlash(filepath.Clean(cfg.Root + string(filepath.Separator) + tArgs[1]))
 
 			if err := SetTemplate(mdc.Template, tArgs[0], fpath); err != nil {
-				c.Errf("template parse error: %v", err)
+				return c.Errf("template parse error: %v", err)
 			}
+
+			mdc.TemplateFiles[tArgs[0]] = fpath
 			return nil
 		}
 	case "templatedir":
 		if !c.NextArg() {
 			return c.ArgErr()
 		}
-		_, err := mdc.Template.ParseGlob(c.Val())
+
+		pattern := c.Val()
+		_, err := mdc.Template.ParseGlob(pattern)
 		if err != nil {
-			c.Errf("template load error: %v", err)
+			return c.Errf("template load error: %v", err)
+		}
+		if c.NextArg() {
+			return c.ArgErr()
 		}
+
+		paths, err := filepath.Glob(pattern)
+		if err != nil {
+			return c.Errf("glob %q failed: %v", pattern, err)
+		}
+		for _, path := range paths {
+			mdc.TemplateFiles[filepath.Base(path)] = path
+		}
+		return nil
+	case "no_cache_template":
```

Comment:
I wonder if we should change this to `cache_templates`, so that the default behavior mirrors that of `templates` directive, which is to not cache them. That would make more sense to me, anyway.